src/parser/ParsedQuery.cpp

```diff
@@ -394,10 +521,43 @@ void ParsedQuery::GraphPattern::toString(std::ostringstream& os,
   os << "}";
 }
 
+// _____________________________________________________________________________
+void ParsedQuery::GraphPattern::recomputeIds(size_t* id_count) {
+  bool allocatedIdCounter = false;
+  if (id_count == nullptr) {
+    id_count = new size_t(0);
```

Comment:
I missed this during the review but we could just use a stack var here and pass a pointer to that to the recursive call only if the parameter was null. This leaves the same variable in the recursive calls unused but that should still be faster than an allocation.